src/Silex/Provider/SwiftmailerServiceProvider.php

```diff
@@ -97,13 +86,34 @@ public function register(Container $app)
             ));
         };
 
-        $app['swiftmailer.transport.eventdispatcher'] = function () {
-            return new \Swift_Events_SimpleEventDispatcher();
+        $app['swiftmailer.transport.eventdispatcher'] = function ($app) {
+            $dispatcher = new \Swift_Events_SimpleEventDispatcher();
+
+            $plugins = $app['swiftmailer.plugins'];
+
+            if (null !== $app['swiftmailer.sender_address']) {
+                $plugins[] = new \Swift_Plugins_ImpersonatePlugin($app['swiftmailer.sender_address']);
+            }
+
+            if (!empty($app['swiftmailer.delivery_addresses'])) {
+                $plugins[] = new \Swift_Plugins_RedirectingPlugin(
+                    $app['swiftmailer.delivery_addresses'],
+                    $app['swiftmailer.delivery_whitelist']
+                );
+            }
+
+            foreach ($plugins as $plugin) {
+                $dispatcher->bindEventListener($plugin);
+            }
+
+            return $dispatcher;
         };
 
+        $app['swiftmailer.plugins'] = array();
```

Comment:
This should not be an array as per your example in the docs, the plugins would not be lazy-loaded. Instead, you should do the same as for `validator.object_initializers` for instance:

```php
$app['swiftmailer.plugins'] = function ($app) {
    return array();
}
```